enterprise/models.py

```diff
@@ -2083,13 +2083,27 @@ def revoke(self):
         """
         Marks this object as revoked and marks the associated EnterpriseCourseEnrollment
         as "saved for later".  This object and the associated EnterpriseCourseEnrollment are both saved.
+
+        TODO: revoke entitlements as well?
         """
+        if self.enterprise_course_enrollment:
+            self.enterprise_course_enrollment.saved_for_later = True
+            self.enterprise_course_enrollment.unenrolled = True
+            self.enterprise_course_enrollment.save()
+
         self.is_revoked = True
-        self.enterprise_course_enrollment.saved_for_later = True
-        self.enterprise_course_enrollment.unenrolled = True
-        self.enterprise_course_enrollment.save()
+        self.save()
 
-        # TODO revoke entitlements as well?
+    def renew(self, **kwargs):
+        """
+        Idmpotently reactivates this enterprise fulfillment source.
+        """
+        if self.enterprise_course_enrollment:
+            self.enterprise_course_enrollment.saved_for_later = False
```

Comment:
Really not sure what saved_for_later means, but I threw it in.  Reviewers, please let me know if this is correct.

Reply:
`saved_for_later` does basically what it says - when true, the enrollment is rendered in the "Saved for later" section of the learner portal dashboard.  I would probably just leave the value unchanged during renewal, but it's not a big deal either way.